pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -185,7 +185,8 @@ type Controller struct {
 
 	knownNodeSet map[string]*v1.Node
 	// per Node map storing last observed health together with a local time when it was observed.
-	nodeHealthMap map[string]*nodeHealthData
+	nodeHealthLock sync.Mutex
+	nodeHealthMap  map[string]*nodeHealthData
```

Comment:
Please encapsulate this to its own structure (this way we will be sure that we changed everything that was needed).
The structure will have get() and set() methods basically (that will be protected with mutex).

Also do it in a separate PR :)

Reply:
ref #82489